src/services/base.service.js

```diff
@@ -51,6 +108,73 @@ const baseSearchById = async (args, resourceType) => {
   return new dataType(result);
 };
 
+/**
+ * Searches using query parameters.
+ * @param {*} args The args from the request.
+ * @param {*} req The Express request object. This is used by the query builder.
+ * @param {*} resourceType The resource type we are searching on.
+ * @param {*} paramDefs Optional parameter definitions for the specific resource types. Specific
+ *                      resource services should call this and pass
+ * @returns Search set result bundle
+ */
+const baseSearch = async (args, { req }, resourceType, paramDefs = {}) => {
+  // grab the schemas for the data type and Bundle to use for response
+  const dataType = resolveSchema(args.base_version, resourceType.toLowerCase());
+  const Bundle = resolveSchema(args.base_version, 'bundle');
+
+  // wipe out params since the 'base_version' here breaks the query building
+  req.params = {};
+
+  // build result bundle. default to an empty result
+  const searchBundle = new Bundle({
+    type: 'searchset',
+    meta: { lastUpdated: new Date().toISOString() },
+    total: 0
+  });
+
+  // build the aggregation query
+  const filter = qb.buildSearchQuery({ req: req, includeArchived: true, parameterDefinitions: paramDefs });
+
+  // if the query builder was able to build a query actually execute it.
+  if (filter.query) {
+    // grab the results from aggregation. has metadata about counts and data with resources
+    const results = (await (await findResourcesWithAggregation(filter.query, resourceType)).toArray())[0];
```

Comment:
Is there a benefit at all to pawning off the `toArray` bit to the mongo controller function instead, to make it always return an array?

This is what we did with the `findResourcesWithQuery` function, so might be worth it for consistency

Reply:
Yeah. I can make it consistent. :)